### ISSUE TITLE
<xutility>: Teach _Ptr_*_cat about contiguous iterators

### ISSUE THREAD

**Issue by miscco** [microsoft/STL#860]
Similar to #819 `_Ptr*_cat` currently only looks at plain pointers. We should expand that to contiguous_iterators

**Comment by miscco** [comment]
@CaseyCarter it is weekend, do some fun stuff

**Comment by miscco** [comment]
Oh this one led me to a beautiful ICE:
```cpp
template<class T> 
inline constexpr bool bark = false;

template<class T, bool = bark<T>> 
inline constexpr bool meow = true;

template<class T> 
inline constexpr bool meow<T*> = meow<T>; 
```

**Comment by StephanTLavavej** [comment]
Did you get a chance to report that ICE on Developer Community for the compiler team to investigate?

**Comment by miscco** [comment]
Yes I already [did](https://developercommunity.visualstudio.com/content/problem/1046737/ice-specialization-of-variable-template.html) and it is already marked as triaged and under consideration

**Comment by AdamBucior** [comment]
This has already been fixed by #1772.
